Release dev-staging changes to dev

curtisupshall opened the pull request:
Changes from `dev-staging` have already been reviewed.

## Links to Jira tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-117
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-226
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-258
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-296
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-297
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-313
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-323
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-324
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-325
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-331
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-334
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-335
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-336
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-337
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-341
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-342
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-343
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-344
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-345
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-347
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-348
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-350
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-352
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-353
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-354
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-355
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-356
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-363
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-371
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-361

curtisupshall committed:
SIMSBIOHUB-226: Survey Prototyping + Observation Map View (#1116)

* Various survey page layout changes
* Adds the ability to see added survey observations in a map view

curtisupshall committed:
Merge branch 'dev' into dev-staging

jeznorth committed:
UI Fixes

curtisupshall committed:
Merge branch 'dev' into dev-staging

NickPhura committed:
Fix issues

curtisupshall committed:
Merge branch 'dev' into dev-staging

curtisupshall committed:
Merge branch 'dev-staging' of github.com:bcgov/biohubbc into dev-staging

NickPhura committed:
Fix pipeline route issues

NickPhura committed:
Fix non-unique key issue

NickPhura committed:
Merge branch 'dev' into dev-staging

jeznorth committed:
SIMSBIOHUB-258 (#1134)

* Minor Fixes / UI Tweaks
* Fix permit component list key
* Rename old sampling method form to sampling strategy

Co-authored-by: Nick Phura <nickphura@gmail.com>

mauberti-bc commented:
![image](https://github.com/bcgov/biohubbc/assets/108430771/592c5d18-3720-45c0-900a-62e19eb05f42)

When you delete an observation row before saving edits, you get a key error on the data grid.

curtisupshall committed:
SIMSBIOHUB-296/313: Import Observations (#1128)

* Add's UIs for uploading CSV files to survey observations
* Adds a new database table to track survey observation CSV submissions, survey_observation_submission
* Alters the survey_observation table to make optional the sampling site, method, and period columns
* Adds service and repo methods for processing observation CSVs

al-rosenthal committed:
SIMSBIOHUB-177: Survey Study Area enhancements  (#1136)

* Survey Study area can now import files with multiple features
* Drawn features can be created and saved to a survey
* New card for displaying and editing map features
* Region selector updated

NickPhura committed:
Merge branch 'dev' into dev-staging

JeremyQuartech committed:
SIMSBIOHUB-336/337 (#1142)

* Adds Sampling Site record count to title area, Example: Sampling Sites (10)
* Adds Observation record count to title area of observations map component and manage observations page, Example: Observations (10)
* Adds 'supplementaryObservationData' section to get survey observations endpoint. This allows observation record counts to be displayed, even if the get observations response becomes paginated or virtualized at a later date

GrahamS-Quartech committed:
SIMSBIOHUB 343 - Incorrect date overlap error in sampling methods (#1144)

* Fixed an issue where the yup schema used for editing sampling methods used the wrong field for a date overlap check.
* Removed an unintended required().
* Added a refresh call when submitting sampling method changes so they are reflected in the UI. Otherwise you will still see the previous value if you go back to the sampling methods view even though the new value did save to the db.

curtisupshall committed:
SIMSBIOHUB-334/350/352 (#1143)

* Added snackbar popup upon successfully saving observations.
* Removed single-click to edit; Fixed bug that made row go into edit mode when clicking row delete icon.
* Added seconds to the observation time

JeremyQuartech committed:
SIMSBIOHUB-323: Remove Ecological Season (#1153)

* Removes all UI references to Ecological Season
* Removes all API references to Ecological Seasons
* Adds DB migration for deprecation comments on Ecological seasons column and xref table
* Removes ecological season from DB seed

GrahamS-Quartech committed:
SIMSBIOHUB 347/341 - Names from shapefiles (#1146)

* Added some additional logic to the API to extract a name and description from the provided shape file when creating sampling sites.
* It will search the keys of the geometry's properties for "name" or "label" and "des" or "desc" or "descr" (exact match), then use the value for the sampling site name and description respectively.
* Added a character limit to the sampling site name field in the edit form.

---------

Co-authored-by: JeremyQuartech <jeremy.kissack@quartech.com>

al-rosenthal committed:
TechDebt: Map container refactor (#1148)

Map container removed in lieu of using the leaflet component 
Map css moved into single component
Removed project location data  

---------

Co-authored-by: Nick Phura <nickphura@gmail.com>
Co-authored-by: JeremyQuartech <jeremy.kissack@quartech.com>
Co-authored-by: jeznorth <twenty15creative@gmail.com>

al-rosenthal committed:
SIMSBIOHUB-353/325: Survey form updates (#1149)

Removed field method and site selection strategies are now required on surveys

KjartanE committed:
SIMSBIOHUB-342:  Delete Sampling Site (#1147)

* Add delete sample site button to menu drop down
* Check for any associated observations to sample site.
* hook up api sample-site delete function.

* SIMSBIOHUB-336/337 (#1142)

* Adds Sampling Site record count to title area, Example: Sampling Sites (10)
* Adds Observation record count to title area of observations map component and manage observations page, Example: Observations (10)
* Adds 'supplementaryObservationData' section to get survey observations endpoint. This allows observation record counts to be displayed, even if the get observations response becomes paginated or virtualized at a later date

* SIMSBIOHUB 343 - Incorrect date overlap error in sampling methods (#1144)

* Fixed an issue where the yup schema used for editing sampling methods used the wrong field for a date overlap check.
* Removed an unintended required().
* Added a refresh call when submitting sampling method changes so they are reflected in the UI. Otherwise you will still see the previous value if you go back to the sampling methods view even though the new value did save to the db.

* SIMSBIOHUB-334/350/352 (#1143)

* Added snackbar popup upon successfully saving observations.
* Removed single-click to edit; Fixed bug that made row go into edit mode when clicking row delete icon.
* Added seconds to the observation time

* Clean up functions and excess data calls

* fix tests

---------

Co-authored-by: JeremyQuartech <123425360+JeremyQuartech@users.noreply.github.com>
Co-authored-by: GrahamS-Quartech <112989452+GrahamS-Quartech@users.noreply.github.com>
Co-authored-by: Curtis Upshall <curtisupshall@gmail.com>

GrahamS-Quartech committed:
SIMSBIOHUB 324 - Multiple Ecological Variables (Intended Outcomes) (#1152)

* Added a migration that removes the intended_outcome_id column from the survey table and shifts the responsibility onto a join table called survey_intended_outcome. This makes it possible to have multiple intended outcomes associated with a single survey.
* Modified the frontend and API to support this change. The dropdown in the add and edit forms for the survey have been replaced with a multiselect. Multiselect will still display the subtext associated with these elements.
* In various cases where a single intended outcome was displayed before, they display as a comma deliminated list instead.
* These items now display with the label of "Ecological Variables" in the frontend survey components.

GrahamS-Quartech committed:
SIMSBIOHUB 335: Tab Through Observation Edit (#1156)

* Modified each custom renderEditCell component of the ObservationTable to manually set hasFocus. This allows the user to Tab forward to the next cell or Shift+Tab back to the previous.
* Wrapped some of the simpler TextField / DatePicker renders in their own functional components so that the same hasFocus pattern could be used.

NickPhura committed:
SIMSBIOHUB-356: Update survey list endpoint to return a subset of survey fields (#1158)

* Replace existing survey list endpoint with index, which returns a subset of survey fields rather than all fields.

NickPhura committed:
SIMSBIOHUB-331: Keycloak Library Updates (#1151)

* Swap out old keycloak libraries for newer more up to date libraries.
* Add RouteGuards tests.
* Add git action to clean PR builds for PRs against non-base branches

al-rosenthal committed:
SIMSBIOHUB-354-355: Study area and Sampling sites on observations map (#1157)

SIMSBIOHUB-354-355: Added sampling site and study area features to the observation map on the survey details page
---------

Co-authored-by: Nick Phura <nickphura@gmail.com>

curtisupshall committed:
SIMSBIOHUB-348: Delete single or many observation records (#1154)

* Added a new ObservationTableContext
* Added an endpont and repository methods to delete one or more observation records
* Added loading button props to the YesNoDialog
* Refactored the observation context to support deleting one or more observation records

GrahamS-Quartech committed:
SIMSBIOHUB-363: Import Survey Area Name and Description from Shapefile (#1161)

* Moved the shapefile name and description extraction to UI. Modified the sampling site POST api to accept custom names and desc from UI. Implemented the same ability in study area submission, now both parts of the form can reference the same logic for extracting this info.

jeznorth committed:
SIMSBIOHUB-334 UI Fixes (#1164)

* Minor UI polish and updating notification content.

JeremyQuartech committed:
SIMSBIOHUB-297: Keycloak Value Case Sensitivity (#1166)

* Removes toLowerCase casting on user_identifier fields from keycloak and stores raw string value in the database regardless of case
* Adds LOWER() sql statements on any relation to user_identifiers and user guids as to make them case-insensitive

KjartanE committed:
SIMSBIOHUB-345: Display Time in Sampling Periods (#1167)

Added time to periods in sampling sites

---------

Co-authored-by: Alfred Rosenthal <alfred.k.rosenthal@gmail.com>

GrahamS-Quartech committed:
SIMSBIOHUB-371: Update Seed to Create Sampling Site and Observation Records (#1165)

* Seed script will now inject a survey sampling site with a corresponding method, period, and three observations.

curtisupshall committed:
SIMSBIOHUB-344: Tech debt: Observation Table Species display (#1160)

* Adds the TaxonomyContext for caching taxonomy names from IDs
* Updates the skeleton loader on the observations table
* Adds backend caching for taxonomy name lookup
* Updates service and repo methods for fetching observations supplementary data
* Fixes a bug with updating the observation count upon deletion
* Fixes a bug where deleted observation records would reappear when going back to the survey page after deletion
* Adds a missing snackbar shown upon deletion
* Adds the count of observtions being deleted in the confirm deletion dialog

al-rosenthal committed:
SIMSBIOHUB-361: Remove all study areas button (#1163)

Added a delete all button to study areas, updated map UI, added layer colors

---------

Co-authored-by: jeznorth <twenty15creative@gmail.com>

al-rosenthal reviewed (approved):
💣

NickPhura committed:
ignore-skip
